nixos/modules/tasks/network-interfaces.nix

```
@@ -506,17 +506,17 @@ in
     networking.fqdnOrHostName = mkOption {
       readOnly = true;
       type = types.str;
-      default = if cfg.domain == null then cfg.hostName else cfg.fqdn;
-      defaultText = literalExpression ''
-        if cfg.domain == null then cfg.hostName else cfg.fqdn
-      '';
+      default =  if (cfg.hostName == "" || cfg.domain == null)
```

```suggestion
      default =  if (cfg.hostName != "" && cfg.domain == null)
```
shouldn't this be like this? otherwise we take an empty hostname as before.

---

Not exactly, the regression occurs when hostname is empty and domain is defined. In that case, I want FqdnOrHostname to return "", not an error (as networking.fqdn does).

---

I find the condition a bit hard to understand when we re-use the empty string hostname in that case.

Also I think this not really covers the original usecase anymore. Before you would always get either a hostname or domain where as empty string probably breaks most usages of it.